angular/src/shared/background-manager.ts

```diff
@@ -117,6 +117,26 @@ export class BackgroundManager {
     return { network, account, accountState, accountHistory };
   }
 
+  async getDerivedKeyFromWalletPath(walletId: string, accountId: string, path: string) {
+    await this.sharedManager.loadPrivateKeys();
+
+    // Get the secret seed.
+    const masterSeedBase64 = this.sharedManager.getPrivateKey(walletId);
+    const masterSeed = Buffer.from(masterSeedBase64, 'base64');
+
+    const wallet = await this.sharedManager.getWallet(walletId);
+    const account = this.sharedManager.getAccount(wallet, accountId);
+    const network = this.sharedManager.getNetwork(account.networkType);
+
+    // Create the master node.
+    const masterNode = HDKey.fromMasterSeed(masterSeed, network.bip32);
+
+    // based on BCIP3 we allow to derive a key only
+    // under the wallet path and it must be hardened keys.
+    var node = masterNode.derive(`m/3'/${account.network}'/${path}`);
```

Comment:
I am aware that is is potentially a security issue and we should discuss it

Reply:
Does path come from website? Don't do that, but you can make it into separate "number" objects, that makes it safer.

Reply:
Something like:
```
{ account: number, index: number }
```

Reply:
no it comes from the handler